### ISSUE TITLE
Problems with .rds file

### ISSUE THREAD

**Issue by cls03** [brianstock/MixSIAR#360]
I've been running models on several computers since they are very heavy, and have been having people save them to .rds files. When the model is originally run, it produces all the necessary (diagnostic, summary, posterior plots, etc). In a few instances, the files were produced but the model was saved to .rds. I am getting error's now when trying to produce the diagnostic and summary information when running the .rds file from my computer- and I need those files. The only changes that are made in the script are that the model isn't being run, and is just being loaded by .rds. Otherwise, no other changes have been made. I have also run MixSIAR models on my computer, so it isn't a clash of my computer vs other computers.

The error that occurs is: 
Error in `[.default`(jags1.mcmc, , v) : incorrect number of dimensions, 
when trying to run 
df.stats <- output_JAGS(jags.siteA_overall, mix, source, output_options). 

My R is 4.3.0 and MixSIAR is version 3.1.12.

Has anyone had a similar issue or know how to resolve it without having to run the model again? I am not super R savy and am not sure what I am missing, but it's pretty urgent. I'd appreciate any help!!

What I do:
I set the working directory, load three files with isotope data, which then produce my isospace, then run the following: 

model_filename <- "MixSIAR_model_siteA_TT_all.txt"
resid_err <- TRUE
process_err <- TRUE
write_JAGS_model(model_filename, resid_err, process_err, mix, source)

jags.siteA_overall <- jags.1

saveRDS(jags.siteA_overall, file = "C://Users//baldg/OneDrive - University of Pisa//PISA//DOTTORATO//siteeisostuf//jagssitee.rds")

I'm then sent the .rds file where I run the three original files, and:

output_options <- list(summary_save = TRUE,                 
                       summary_name = "summary_statistics_siteA_TT", 
                       sup_post = FALSE,                    
                       plot_post_save_pdf = TRUE,           
                       plot_post_name = "posterior_density_siteA_TT",
                       sup_pairs = FALSE,             
                       plot_pairs_save_pdf = TRUE,    
                       plot_pairs_name = "pairs_plot_siteA_TT",
                       sup_xy = TRUE,           
                       plot_xy_save_pdf = FALSE,
                       plot_xy_name = "xy_plot_siteA_TT",
                       gelman = TRUE,
                       heidel = FALSE,  
                       geweke = TRUE,   
                       diag_save = TRUE,
                       diag_name = "diagnostics_siteA_TT",
                       indiv_effect = FALSE,       
                       plot_post_save_png = TRUE, 
                       plot_pairs_save_png = TRUE,
                       plot_xy_save_png = TRUE,
                       diag_save_ggmcmc = TRUE,
                       return_obj = TRUE)

output_options <- list(summary_save = TRUE,                 
                       summary_name = "summary_statistics_siteA_TT", 
                       sup_post = FALSE,                    
                       plot_post_save_pdf = TRUE,           
                       plot_post_name = "posterior_density_siteA_TT",
                       sup_pairs = FALSE,             
                       plot_pairs_save_pdf = TRUE,    
                       plot_pairs_name = "pairs_plot_siteA_TT",
                       sup_xy = TRUE,           
                       plot_xy_save_pdf = FALSE,
                       plot_xy_name = "xy_plot_siteA_TT",
                       gelman = TRUE,
                       heidel = FALSE,  
                       geweke = TRUE,   
                       diag_save = TRUE,
                       diag_name = "diagnostics_siteA_TT",
                       indiv_effect = FALSE,       
                       plot_post_save_png = TRUE, 
                       plot_pairs_save_png = TRUE,
                       plot_xy_save_png = TRUE,
                       diag_save_ggmcmc = TRUE,
                       return_obj = TRUE)

output_posteriors(jags.siteA_overall, mix, source, output_options)

df.stats <- output_JAGS(jags.siteA_overall, mix, source, output_options)

Error in `[.default`(jags1.mcmc, , v) : incorrect number of dimensions
6.
NextMethod("[")
5.
`[.mcmc`(jags1.mcmc, , v)
4.
jags1.mcmc[, v]
3.
as.mcmc.list(x)
2.
coda::gelman.diag(jags1.mcmc[, v])
1.
output_JAGS(jags.siteA_overall, mix, source, output_options)

**Comment by brianstock** [comment]
Hard to tell without all the code. Can you reproduce this error using one
of the datasets distributed with mixsiar (eg wolves) and test run length?

**Comment by cls03** [comment]
Hi Brian, 

Thanks for the response. I was getting the same error with the wolves test run, but I managed a way around it. I got the code for the output_JAGS function, renamed it, and ran it on my computer. I then uploaded the .rds file and ran the output_options, output_posteriors and df.stats with the newly named function and it prduced the diagnostics and summary.

I have no idea why it worked, but thankfully did.

Thanks though!
